tests/test_factory_fields.py

```diff
@@ -161,3 +161,23 @@ def caption(cls, is_long: bool) -> str:
         assert result.caption == "this was really long for me"
     else:
         assert result.caption == "just this"
+
+
+@pytest.mark.parametrize(
+    "factory_field",
+    [
+        Use(lambda: "foo"),
+        PostGenerated(lambda: "foo"),
+        Require(),
+        Ignore(),
+    ],
+)
+def test_non_existing_model_fields_raises(factory_field: Union[Use, PostGenerated, Require, Ignore]) -> None:
+    class NoFieldModel(BaseModel):
+        pass
+
+    with pytest.raises(
+        ConfigurationException,
+        match="unknown_field is declared on the factory NoFieldModelFactory but it is not part of the model NoFieldModel",
+    ):
+        ModelFactory.create_factory(NoFieldModel, bases=None, **{"unknown_field": factory_field})
```

Comment:
```suggestion
        ModelFactory.create_factory(NoFieldModel, bases=None, unknown_field=factory_field)
```

This should also work right?

Reply:
Yep I initially made it more complex with the dict being defined in the parametrize definition, thanks for spotting :+1: